collectors/python.d.plugin/dns_query_time/dns_query_time.chart.py

```diff
@@ -85,7 +85,7 @@ def dns_req(ns, t, q):
 
         try:
             resp = dns.query.udp(request, ns, timeout=t)
-            if not (resp.rcode == dns.rcode.NOERROR or resp.answer):
+            if (resp.rcode == dns.rcode.NOERROR and not resp.answer):
```

Comment:
wait wait

`resp.rcode == dns.rcode.NOERROR` is OK

lets do

```python
if resp.rcode == dns.rcode.NOERROR and resp.answer: 
    ok ...
else:
    not ok...
```

Reply:
we good only if `resp.rcode == dns.rcode.NOERROR` and `resp.answer`, right?

Reply:
got it, pushing another commit now

Reply:
hmm, doesn't seem to work that way

simple code snipet:
```
import dns.query

try:
    qname = dns.name.from_text('www.google.com')
    q = dns.message.make_query(qname, dns.rdatatype.A)
    r = dns.query.udp(q, '8.8.8.8')
    resp.rcode == dns.rcode.NOERROR
except dns.exception.Timeout:
    print('failure')
except:
    print('fallback')
```

Reply:
```python
    def dns_req(ns, t, q):
        domain = dns.name.from_text(choice(domains))
        request = dns.message.make_query(domain, dns.rdatatype.A)
        query_time = -100

        try:
            resp = dns.query.udp(request, ns, timeout=t)
            if resp.rcode == dns.rcode.NOERROR and resp.answer:
                query_time = resp.time * 100000
        except dns.exception.DNSException:
            pass
        finally:
            name = '_'.join(['ns', ns.replace('.', '_')]
            q.put({name: query_time})
```

Reply:
multiplying by 10000 gives _huge_ numbers, since it's ms to sec 1000 should be correct and looks good on the graphs I've been using to validate. resp.rcode needs parenthesis - both should be in the commit being validated now.

Reply:
> multiplying by 10000 gives huge numbers

multiply by `100000` (not 10000) in the code and add divisor 100 to chart